src/Microsoft.ML.Data/Data/ITransposeDataView.cs

```diff
@@ -14,46 +14,30 @@ namespace Microsoft.ML.Data
-    /// <see cref="IDataView"/>.
+    /// cursor via an <see cref="SlotCursor"/> returned by <see cref="GetSlotCursor(int)"/>
+    /// (naturally, as opposed to row-by-row through an <see cref="RowCursor"/>). This interface
+    /// is intended to be implemented by classes that want to provide an option for an alternate
+    /// way of accessing the data stored in a <see cref="IDataView"/>.
     ///
-    /// The interface only advertises that columns may be accessible in slot-wise fashion. A column
-    /// is accessible in this fashion iff <see cref="TransposeSchema"/>'s
-    /// <see cref="ITransposeSchema.GetSlotType"/> returns a non-null value.
+    /// The interface only advertises that columns may be accessible in slot-wise fashion. The i-th column
+    /// is accessible in this fashion iff <see cref="TransposeSlotTypes"/>[i] is not <see langword="null"/>
+    /// and <see cref="TransposeSlotTypes"/>[i] is not <see langword="null"/>.
     /// </summary>
     [BestFriend]
     internal interface ITransposeDataView : IDataView
     {
-        /// <summary>
-        /// An enhanced schema, containing information on the transposition properties, if any,
-        /// of each column. Note that there is no contract or suggestion that this property
-        /// should be equal to <see cref="IDataView.Schema"/>.
-        /// </summary>
-        ITransposeSchema TransposeSchema { get; }
-
         /// <summary>
         /// Presents a cursor over the slots of a transposable column, or throws if the column
         /// is not transposable.
         /// </summary>
         SlotCursor GetSlotCursor(int col);
-    }
 
-    /// <summary>
-    /// The transpose schema returns the schema information of the view we have transposed.
-    /// </summary>
-    [BestFriend]
-    internal interface ITransposeSchema : ISchema
-    {
         /// <summary>
-        /// Analogous to <see cref="ISchema.GetColumnType"/>, except instead of returning the type of value
-        /// accessible through the <see cref="RowCursor"/>, returns the item type of value accessible
-        /// through the <see cref="SlotCursor"/>. This will return <c>null</c> iff this particular
-        /// column is not transposable, that is, it cannot be viewed in a slotwise fashion. Observe from
-        /// the return type that this will always be a vector type. This vector type should be of fixed
-        /// size and one dimension.
+        /// <see cref="TransposeSlotTypes"/>[i] specifies the type of all values at the i-th column of <see cref="IDataView"/>.
+        /// For example, if <see cref="IDataView.Schema"/>[i] is a scalar float column, then <see cref="TransposeSlotTypes"/>[i]
+        /// may return a <see cref="VectorType"/> whose <see cref="VectorType.ItemType"/> field is <see cref="NumberType.R4"/>.
+        /// If the i-th column can't be iterated column-wisely, <see cref="TransposeSlotTypes"/>[i] may be <see langword="null"/>.
```

Comment:
>cref="TransposeSlotTypes"/>[i] may be <see langword="null"/> [](start = 69, length = 60)

Also apparently, from what I see of the usage, even the array *itself* might be `null`. The more I read of the new code the more I think we got it right the first time with a method, not exposing some collection that we have to test whether it is null or not before we even think about querying... yeah.

Reply:
Just replaced this with a function, `GetSlotType(int col)`. Many thanks.

---
In reply to: [248780062](https://github.com/dotnet/machinelearning/pull/2111#discussion_r248780062) [](ancestors = 248780062)